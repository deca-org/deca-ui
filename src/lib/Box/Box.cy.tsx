import { Test } from '../Utils';
import _cyp from '../../../cypress';
import React from 'react';
import Box from './Box';

describe('components/box', () => {
  const selector = '[data-testid="test.box"]';
  it('fontFamily', () => {
    cy.mount(<Box data-testid="test.box">Text</Box>);
    cy.get(selector).should('have.css', 'font-family', Test.font('normal'));
  });
});
