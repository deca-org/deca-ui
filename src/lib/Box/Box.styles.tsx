import { styled } from '@lib/Theme/stitches.config';

export const StyledBox = styled('div', {
  fontFamily: '$normal',
});
