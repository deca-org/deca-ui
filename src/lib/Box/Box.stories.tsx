import Text from '@lib/Text';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import Box from './Box';

export default {
  title: 'Box',
  component: Box,
} as ComponentMeta<typeof Box>;

const Template: ComponentStory<typeof Box> = (args) => <Box {...args} />;

const showcaseText = 'Almost before we knew it, we had left the ground.';

export const Default = Template.bind({});

Default.args = {
  css: {
    br: '$lg',
    size: '$25',
    linearGradient: '20deg, $pink900, $red500',
  },
};

export const WithChildren = () => (
  <Box
    css={{
      fontFamily: '$normal',
      bg: '$primary',
      color: '$white',
      width: 'calc($35 * 1.75)',
      p: '$4',
      br: '$lg',
      linearGradient: '20deg, $blue900, $blue500',
    }}
  >
    <Text as="h4">{showcaseText}</Text>
    <Text>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed eleifend
      rhoncus ligula, eget porta enim aliquam nec. Suspendisse ultrices lorem
      lobortis feugiat tempor.
    </Text>
  </Box>
);
