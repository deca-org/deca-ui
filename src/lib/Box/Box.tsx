import { CSS } from '@lib/Theme/stitches.config';
import {
  __DEV__,
  PolymorphicRef,
  PolymorphicComponentPropWithRef,
} from '@lib/Utils';
import clsx from 'clsx';
import React from 'react';

import { StyledBox } from './Box.styles';

/**
 * The Box component serves as a wrapper component
 */
interface Props {
  /**
   * Override default CSS style.
   */
  css?: CSS;
  /**
   * The content of the component.
   */
  children?: React.ReactNode | undefined;
  /**
   * ClassName applied to the component.
   * @default ''
   */
  className?: string;
}

export type BoxProps<T extends React.ElementType> =
  PolymorphicComponentPropWithRef<T, Props>;

export type BoxComponent = (<C extends React.ElementType = 'div'>(
  props: BoxProps<C>
) => React.ReactElement | null) & { displayName?: string };

const Box: BoxComponent = React.forwardRef(
  <T extends React.ElementType = 'div'>(
    { as, css, children, className = '', ...boxProps }: BoxProps<T>,
    ref?: PolymorphicRef<T>
  ) => {
    const preClass = 'decaBox';

    return (
      <StyledBox
        as={as}
        css={css}
        className={clsx(className, `${preClass}-root`)}
        ref={ref}
        {...boxProps}
      >
        {children}
      </StyledBox>
    );
  }
);

if (__DEV__) {
  Box.displayName = 'DecaUI.Box';
}

export default Box;
