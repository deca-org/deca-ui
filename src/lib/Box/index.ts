import Box from './Box';
export * from './Box';
export { Box };
export default Box;
