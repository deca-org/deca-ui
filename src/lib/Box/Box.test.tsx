import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';

import Box from './Box';

describe('components/Box', () => {
  it('renders content', () => {
    const { getByText } = render(<Box>Children</Box>);
    expect(getByText('Children')).toBeInTheDocument();
  });
  it('matches snapshot', () => {
    const { asFragment } = render(<Box></Box>);
    expect(asFragment()).toMatchSnapshot();
  });
});
