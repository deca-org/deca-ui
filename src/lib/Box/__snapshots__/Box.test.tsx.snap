// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Box matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="c-dTHfCl decaBox-root"
  />
</DocumentFragment>
`;
